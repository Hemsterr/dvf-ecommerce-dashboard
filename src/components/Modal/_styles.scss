.modal {
  @include absolute-position(50%, auto, auto, 50%);
  border: solid pxToRem(10px) $shuttle-gray;
  background: $white;
  width: pxToRem(600px);
  height: auto;
  z-index: 5;
  align-items: center;
  justify-content: center;
  transform: translate(-50%, -50%);

  &__default {
    background: $white;
    text-align: center;

    .modal {
      &__header {
        display: none;
      }

      &__dialog {
        padding: pxToRem(35px);
      }
    }
  }

  &__icon {
    width: pxToRem(44px);
    height: pxToRem(28px);
    background: url('../../assets/icons/close-icon.png');
    background-size: cover;
    display: block;
    margin-left: pxToRem(8px);
  }

  &__dialog {
    padding: pxToRem(10px) 0 pxToRem(35px);
    height: 100%;
  }

  &__header {
    display: flex;
    justify-content: flex-end;
    padding-bottom: pxToRem(10px);
    cursor: pointer;
  }

  &__body {
    padding-top: pxToRem(15px);
    padding-bottom: pxToRem(15px);
  }

  &__title {
    @include text-size(pxToRem(14px), pxToRem(3.5px), pxToRem(17px));
    color: $alabaster;
    font-family: $font-base-futura;
    text-align: center;
    text-transform: uppercase;
  }

  &__backdrop {
    @include absolute-position(0, 0, 0, 0);
  }
}

.modal-wrapper {
  display: flex;
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  align-items: center;
  justify-content: center;
  z-index: 4;
  overflow: hidden;
}

@media only screen and (max-width: $screen-sm) {
  .modal {
    width: 85.7%;
    top: 47%;
  }

  .modal-wrapper {
    display: flex;
    position: fixed;
    top: 50px;
    left: 0;
    bottom: 0;
    right: 0;
    align-items: center;
    justify-content: center;
    z-index: 4;
    overflow: hidden;
  }
}

@media only screen and (max-width: $screen-xs) {
  .modal  {

    &__body {
      padding-top: 0;
    }

    &__dialog {
      padding: pxToRem(10px) 0;
    }
  }
}
