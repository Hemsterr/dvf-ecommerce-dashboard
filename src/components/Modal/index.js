// @flow
// Lib
import * as React from 'react'

type Props = {
  handleCloseModal: Function,
  children: React.Node,
  title: string,
  className: string,
}

class Modal extends React.PureComponent<Props> {
  static defaultProps = {
    title: '',
  }

  render() {
    const {
      handleCloseModal, children, title, className
    } = this.props

    return (
      <div className="modal-wrapper">
        <div className={`modal ${className}`}>
          <div className="modal__dialog">
            <div
              className="modal__header"
              role="presentation"
              onClick={handleCloseModal}
            >
              <div className="modal__icon" />
            </div>
            <p className="modal__title">{title}</p>
            <div className="modal__body">{children}</div>
          </div>
        </div>
        <div
          role="presentation"
          className="modal__backdrop"
          onClick={handleCloseModal}
        />
      </div>
    )
  }
}

export default Modal
