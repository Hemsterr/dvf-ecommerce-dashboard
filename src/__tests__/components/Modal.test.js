// Libs
import React from 'react'
import { shallowToJson } from 'enzyme-to-json'
import { shallow } from 'enzyme'

// Component
import Modal from '../../components/Modal'

const props = {
  handleCloseModal: jest.fn(),
  children: '',
  title: 'Appointments detail',
  className: 'isUpcoming',
}

const wrapper = (props = {}) => {
  const component = shallow(<Modal {...props} />)

  return component
}

describe('Modal component', () => {
  const wrapperWithProps = wrapper(props)

  // Snapshot
  test('should render correctly Modal component default props', () => {
    expect(shallowToJson(wrapperWithProps)).toMatchSnapshot()
  })

  // Testing props
  describe('Handle with props', () => {
    test('should be rendered default props', () => {
      expect(wrapperWithProps.find('p').length).toBe(1)
    })

    test('should be rendered title modal', () => {
      expect(wrapperWithProps.find('p').props().children).toEqual(
        'Appointments detail'
      )
    })
  })
})
