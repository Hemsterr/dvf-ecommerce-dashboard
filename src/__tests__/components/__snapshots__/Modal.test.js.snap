// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should render correctly Modal component default props 1`] = `
<div
  className="modal-wrapper"
>
  <div
    className="modal isUpcoming"
  >
    <div
      className="modal__dialog"
    >
      <div
        className="modal__header"
        onClick={[MockFunction]}
        role="presentation"
      >
        <div
          className="modal__icon"
        />
      </div>
      <p
        className="modal__title"
      >
        Appointments detail
      </p>
      <div
        className="modal__body"
      />
    </div>
  </div>
  <div
    className="modal__backdrop"
    onClick={[MockFunction]}
    role="presentation"
  />
</div>
`;
